Runtime/Core/AxisState.cs

```diff
@@ -199,6 +205,28 @@ public void SetInputAxisProvider(int axis, IInputAxisProvider provider)
         /// <b>false</b> otherwise</returns>
         public bool Update(float deltaTime)
         {
+            // Update only once per frame
+            if (Time.frameCount == m_LastUpdateFrame)
+                return false;
+            m_LastUpdateFrame = Time.frameCount;
+
+            // Cheating: we want the render frame time, not the fixed frame time
+            if (CinemachineCore.UniformDeltaTimeOverride < 0)
+            {
+                if (deltaTime >= 0 && m_LastUpdateTime != 0)
+                    deltaTime = Time.time - m_LastUpdateTime;
+                m_LastUpdateTime = Time.time;
+            }
+            else
+            {
+                // CinemachineCore.UpdateVirtualCamera can change deltaTime to catch up,
+                // but that should not happen when CinemachineCore.UniformDeltaTimeOverride is set.
+                deltaTime = CinemachineCore.UniformDeltaTimeOverride; 
+                // TODO: deltaTime should not be adjusted in CinemachineCore.UpdateVirtualCamera when
+                // TODO: CinemachineCore.UniformDeltaTimeOverride is set?
+            }
```

Comment:
Tests failed, because they use `CinemachineCore.UniformDeltaTimeOverride` (deltaTime = 0.1f). But this part could overwrite the `deltaTime` value.

--------

While looking at the code, I saw that deltaTime can also be overwritten/adjusted to catch up with frames when update method is smart (fixedSmart or smart). I think, when `UniformDeltaTimeOverride` is set, then `deltaTime` should not be adjusted. Is that true?

If yes, then we should take care of that in `CinemachineCore.UpdateVirtualCamera()`, instead of inside the `else` block here.

Reply:
@glabute

Reply:
Nice catch.  I think CinemachineCore.UpdateVirtualCamera() is ok the way it is.  UniformDeltaTimeOverride  will still be respected on a per-frame basis.